### PR TITLE
rustc_target: Mark UEFI targets as `is_like_windows`/`is_like_msvc`

### PR DESCRIPTION
**Pull request by petrochenkov**
And document what `is_like_windows` and `is_like_msvc` actually mean in more detail.

Addresses FIXMEs left from https://github.com/rust-lang/rust/pull/71030.
r? @nagisa

### CODE REVIEW

**Comment by petrochenkov** [comment]
cc @dvdhrm

File: compiler/rustc_target/src/spec/mod.rs [hunk 1]
```diff
@@ -818,10 +818,22 @@ pub struct TargetOptions {
...
+    /// This is a combination of several more specific properties represented as a single flag:
+    ///   - The target uses a Windows ABI,
+    ///   - uses PE/COFF as a format for object code,
+    ///   - uses Windows-style dllexport/dllimport for shared libraries,
+    ///   - uses import libraries and .def files for symbol exports,
+    ///   - executables support setting a subsystem.
     pub is_like_windows: bool,
+    /// Whether the target is like MSVC.
+    /// This is a combination of several more specific properties represented as a single flag:
+    ///   - The target has all the properties from `is_like_windows`,
```
Comment (nagisa) [inline compiler/rustc_target/src/spec/mod.rs:831]: unclear if setting this implies the `is_like_windows`.

**Comment by nagisa** [comment]
SGTM r=me

**Comment by dvdhrm** [comment]
Thanks for digging this up and adding the comments! Sounds all good to me!

**Pull request merged**